eclipse/che
# Fixup SCM URL matching sometimes fails

### What does this PR do?
It is reported that sometimes user secrets containing SCM tokens are ignored. Investigation showed that SCM URL-s in factory and in secrets compared directly, wichi is not good, since them may differ by trailing slash. This PR fixes the situation and 
now URls are compared without the influence of the trailing slash 

PS. During testing, PAT validation bug  for GitLab found. So it is fixes this bug as well.

### Screenshot/screencast of this PR



### What issues does this PR fix or reference?



### How to test this PR?
1) Create new/edit existing secret with the personal access token, containing `scm-url` annotation with trailing slash, as follows:

```
che.eclipse.org/scm-url: 'https://gitlab.apps.cluster-max-fbfa.max-fbfa.example.opentlc.com/'
```
 2) Run the private GitLab project factory from that server. Error in accessing devfile will be shown. 
 3) Deploy che server image `quay.io/skabashn/che-server:che19752`
 4) After the fix, the factory should run without an error.


### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)


### Reviewers

Reviewers, please comment how you tested the PR when approving it.
